src/main/java/com/refinedmods/refinedstorage/datageneration/BlockModelGenerator.java

```diff
@@ -0,0 +1,236 @@
+            } else {
+                return models.createCubeCutoutModel(
+                    "block/" + getBlockName(block, color) + "/" + color,
+                    BOTTOM,
+                    BOTTOM,
+                    getRL(block, color, "top"),
+                    getRL(block, color, "cutouts/top_" + color),
+                    getRL(block, color, "side"),
+                    getRL(block, color, "cutouts/side_" + color),
+                    getRL(block, color, "side"),
+                    getRL(block, color, "cutouts/side_" + color),
+                    getRL(block, color, "side"),
+                    getRL(block, color, "cutouts/side_" + color),
+                    getRL(block, color, "side"),
+                    getRL(block, color, "cutouts/side_" + color));
+            }
+        }, 180);
+    }
+
+    private void genControllerModel(Block block, Block modelBlock, DyeColor color) {
+        models.simpleBlockStateModel(block, state -> {
+            if (state.get(ControllerBlock.ENERGY_TYPE).equals(ControllerBlock.EnergyType.OFF)) {
+                return models.createCubeAllCutoutModel(
+                    "block/" + getBlockName(modelBlock, color) + "/off",
+                    getRL(modelBlock, color, "off"),
+                    getRL(modelBlock, color, "off"),
+                    getRL(modelBlock, color, "cutouts/off")
+                );
+            } else if (state.get(ControllerBlock.ENERGY_TYPE).equals(ControllerBlock.EnergyType.NEARLY_OFF)) {
+                return models.createControllerNearlyCutoutModel(
+                    "block/" + getBlockName(modelBlock, color) + "/nearly_off",
+                    getRL(modelBlock, color, "off"),
+                    getRL(modelBlock, color, "on"),
+                    getRL(modelBlock, color, "cutouts/nearly_off"),
+                    getRL(modelBlock, color, "cutouts/nearly_off_gray"));
+            } else if (state.get(ControllerBlock.ENERGY_TYPE).equals(ControllerBlock.EnergyType.NEARLY_ON)) {
+                return models.createControllerNearlyCutoutModel(
+                    "block/" + getBlockName(modelBlock, color) + "/nearly_on",
+                    getRL(modelBlock, color, "off"),
+                    getRL(modelBlock, color, "on"),
+                    getRL(modelBlock, color, "cutouts/nearly_on"),
+                    getRL(modelBlock, color, "cutouts/nearly_on_gray"));
+            } else {
+                return models.createCubeAllCutoutModel(
+                    "block/" + getBlockName(modelBlock, color) + "/" + color,
+                    getRL(modelBlock, color, "off"),
+                    getRL(modelBlock, color, "on"),
+                    getRL(modelBlock, color, "cutouts/" + color));
+            }
+        });
+    }
+
+    private void genNorthCutoutModel(Block block, DyeColor color) {
+        models.horizontalRSBlock(block, state -> {
+                if (!state.get(NetworkNodeBlock.CONNECTED)) {
```

Comment:
Pretty sure the indentation is off here, should be 4 not 8? (Also in all the other methods probably)